packages/wonder-blocks-modal/components/modal-launcher/index.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+/**
+ * This component enables you to launch a modal, covering the screen.
+ *
+ * Children have access to `openModal` function via the function-as-children
+ * pattern, so one common use case is for this component to wrap a button:
+ *
+ * <ModalLauncher modal={<TwoColumnModal ... />}>
+ *     {({openModal}) => <button onClick={openModal}>Learn more</button>}
+ * </ModalLauncher>
+ *
+ * The actual modal itself is constructed separately, using a layout component
+ * like StandardModal, OneColumnModal, or TwoColumnModal, and is provided via
+ * the `modal` prop.
+ */
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+import Color from "wonder-blocks-color";
+import {View} from "wonder-blocks-core";
+
+import ModalLauncherPortal from "./modal-launcher-portal.js";
+
+type Props = {
+    // The modal to render.
+    //
+    // The modal will be rendered inside of a container whose parent is
+    // document.body. This allows us to use ModalLauncher within menus and
+    // other components that clip their content. If the modal needs to close
+    // itself by some other means than tapping the backdrop or the default
+    // close button a render callback can be passed. The closeModal function
+    // provided to this callback can be called to close the modal.
+    //
+    // Note: Don't call `closeModal` while rendering! It should be used to
+    // respond to user intearction, like `onClick`.
+    modal: React.Node | (({closeModal: () => void}) => React.Node),
```

Comment:
I'm wondering if we should ever allow devs to specify the modal directly, instead of a function. Specifying it directly is almost always an anti-pattern as it means that we're not async-loading the modal, whereas with the callback we could be. Another option that we may want to consider here is something like: `modal={() => import("...")}` for handling the most common case.

Reply:
If we're expecting devs to use this component directly then we should be trying to get them to async load the modals, if at all possible. No point in having to load all of the modal-related logic if we're just waiting for the loader to activate it, right?

Reply:
Hmm makes sense. @kevinbarabash what do you think?